sdk/servicebus/service-bus/src/receivers/sessionReceiver.ts

```diff
@@ -404,6 +414,47 @@ export class ServiceBusSessionReceiverImpl implements ServiceBusSessionReceiver
     return retry<ServiceBusReceivedMessage[]>(config);
   }
 
+  async deleteMessages(options: DeleteMessagesOptions): Promise<number> {
+    this._throwIfReceiverOrConnectionClosed();
+
+    const deleteMessagesOperationPromise = (): Promise<number> => {
+      return this._context
+        .getManagementClient(this.entityPath)
+        .deleteMessages(options.maxMessageCount, options?.beforeEnqueueTime, this.sessionId, {
+          ...options,
+          associatedLinkName: this._messageSession.name,
+          requestName: "deleteMessages",
+          timeoutInMs: this._retryOptions.timeoutInMs,
+        });
+    };
+    const config: RetryConfig<number> = {
+      operation: deleteMessagesOperationPromise,
+      connectionId: this._context.connectionId,
+      operationType: RetryOperationType.management,
+      retryOptions: this._retryOptions,
+      abortSignal: options?.abortSignal,
+    };
+    return retry<number>(config);
+  }
+
+  async purgeMessages(options?: PurgeMessagesOptions): Promise<number> {
+    let deletedCount = await this.deleteMessages({
+      maxMessageCount: MaxDeleteMessageCount,
+      beforeEnqueueTime: options?.beforeEnqueueTime,
+    });
+    if (deletedCount === MaxDeleteMessageCount) {
```

Comment:
I did something slightly different in my "PurgeMessages" example. Basically, I just keep calling DeleteMessages until it returns 0 as a count.

Ie, if they delete less a page it's not a total guarantee that there isn't anything more left.

Reply:
yeah see my above comment where I did checking of 0.  This is related to the service issue we were discussing earlier in team channel and the service is going to fix to ensure maxMessageCount messages are deleted unless there are less messages left.